odoo/addons/base/transpiler/sourcemap_generator.py

```diff
@@ -0,0 +1,191 @@
+class SourceMapGenerator:
+    """
+    The SourceMapGenerator will create the sourcemap that maps the js files to the asset bundle.
+
+    What is a sourcemap ? (https://developer.mozilla.org/en-US/docs/Tools/Debugger/How_to/Use_a_source_map)
+    In brief: a source map is what make possible to debug your processed/compiled/minified code as your were debuging
+    the the original, non-altered source code. It is a file that provides a mapping original <=> processed for the
+    browser to read.
+
+    This implementation of the SourceMapGenerator is a translation and adaptation of this implementation
+    in js https://github.com/mozilla/source-map. For performance purposes, we have removed all unnecessary
+    functions/steps for our use case. This simpler version does a line by line mapping, with the ability to
+    add offsets at the start and end of a file. (when we have to add comments on top a transpiled file by example).
+    """
+    def __init__(self, args={}):
+        self._file = args.get("file", None)
+        self._sourceRoot = args.get("sourceRoot", None)
+        self._sources = {}
+        self._mappings = []
+        self._sourcesContents = {}
+        self._version = 3
+        self.base64vlq_encode = memorize(base64vlq())
```

Comment:
Could use `functools.lru_cache(maxsize=None)` from the stdlib instead of a custom memo decorator.

Is this done on a per-instance basis because there is low/no chance this will be useful across sourcemaps? Otherwise the encoder should be decorated directly.